### PR TITLE
feat(amplitude-js): Add cookieStorage. Closes #57387

### PR DESCRIPTION
**Pull request by danmana**
Closes #57387 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - see https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/57387#discussioncomment-1729207
  - https://github.com/amplitude/Amplitude-JavaScript/blob/v8.13.0/src/amplitude-client.js#L41
  - https://github.com/amplitude/Amplitude-JavaScript/blob/v8.13.0/src/cookiestorage.js#L25
  - https://github.com/amplitude/Amplitude-JavaScript/blob/v8.13.0/src/cookie.js#L104
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## amplitude-js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/amplitude-js@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'amplitude-js'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. VERSION
1. BROWSER
1. CPU
1. DEVICE

<details>
<summary>as well as these 3 other properties...</summary>
<p>ENGINE, OS, UAParser</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e4031032488f94cf1843c5b9621766c8f02a0d3e
</p>

File: types/amplitude-js/index.d.ts [hunk 1]
```diff
@@ -127,6 +129,21 @@ export class AmplitudeClient {
     Revenue: typeof Revenue;
 }
 
+export interface CookieStorageOptions {
+    expirationDays?: number | undefined;
+    domain?: string | undefined;
+    secure?: boolean | undefined;
+    sameSite?: 'Lax' | 'Strict' | 'None' | undefined;
```
Comment (peterblazejewicz) [inline types/amplitude-js/index.d.ts:136]: should this be in `BaseCookieOptions` and `CookieStorageOptions` should extend it? (no need for BaseCookie)

Comment (danmana) [reply types/amplitude-js/index.d.ts:140]: I'm sorry, I don't think I understand. Do you mean to reuse the typing from `Config.sameSiteCookie`? We don't have any `BaseCookie` typings ...

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by danmana** [comment]
Ready to merge

**Pull request merged**